ote_cli/ote_cli/tools/explain.py

```diff
@@ -0,0 +1,166 @@
+        required=True,
+        help="Comma-separated paths to explain data folders.",
+    )
+    parser.add_argument(
+        "--save-explanation-to",
+        default="saliency_dump",
+        help="Output path for explanation images.",
+    )
+    parser.add_argument(
+        "--load-weights",
+        required=True,
+        help="Load only weights from previously saved checkpoint",
+    )
+    parser.add_argument(
+        "--explain-algorithm",
+        default="EigenCAM",
+        help=f"Explain algorithm name, currently support {SUPPORTED_EXPLAIN_ALGORITHMS}",
+    )
+    parser.add_argument(
+        "-w",
+        "--weight",
+        type=float,
+        default=0.5,
+        help="weight of the saliency map when overlaying the saliency map",
+    )
+    add_hyper_parameters_sub_parser(parser, hyper_parameters, modes=("INFERENCE",))
+
+    return parser.parse_args(), template, hyper_parameters
+
+
+def main():
+    """
+    Main function that is used for model explanation.
+    """
+
+    # Dynamically create an argument parser based on override parameters.
+    args, template, hyper_parameters = parse_args()
+    # Get new values from user's input.
+    updated_hyper_parameters = gen_params_dict_from_args(args)
+    # Override overridden parameters by user's values.
+    override_parameters(updated_hyper_parameters, hyper_parameters)
+
+    hyper_parameters = create(hyper_parameters)
+
+    # Get classes for Task, ConfigurableParameters and Dataset.
+    if args.load_weights.endswith(".pth"):
+        if is_checkpoint_nncf(args.load_weights):
+            task_class = get_impl_class(template.entrypoints.nncf)
+        else:
+            task_class = get_impl_class(template.entrypoints.base)
+    else:
+        raise ValueError(f"Unsupported file: {args.load_weights}")
+
+    environment = TaskEnvironment(
+        model=None,
+        hyper_parameters=hyper_parameters,
+        label_schema=read_label_schema(args.load_weights),
+        model_template=template,
+    )
+
+    environment.model = read_model(
+        environment.get_model_configuration(), args.load_weights, None
+    )
+
+    task = task_class(task_environment=environment)
+
+    if args.explain_algorithm not in SUPPORTED_EXPLAIN_ALGORITHMS:
+        raise NotImplementedError(
+            f"{args.explain_algorithm} currently not supported. \
+            Currently only support {SUPPORTED_EXPLAIN_ALGORITHMS}"
+        )
+
+    image_files = get_image_files(args.explain_data_roots)
+    explain_dataset = get_explain_dataset_from_filelist(image_files)
+
+    if not os.path.exists(args.save_explanation_to):
+        os.makedirs(args.save_explanation_to)
+
+    saliency_maps = task.explain(
+        explain_dataset.with_empty_annotations(),
+        InferenceParameters(
+            is_evaluation=True,
+            explainer=args.explain_algorithm,
+        ),
+    )
+
+    for img, saliency_map, (_, fname) in zip(
+        explain_dataset, saliency_maps, image_files
```

Comment:
You do not need to iterate over `explain_dataset`. `saliency_maps` is a dataset that contains both init image and `saliency_map` as metadata - please you it for iteration.

Reply:
fixed